File: csharp/bob/Bob.cs
using System;
using System.Text.RegularExpressions;

public static class Bob
{
    public static string Response(string statement)
    {
        var TrimmedStatement = statement.Trim();
        var Silent = isSilent(TrimmedStatement);
        var Shouting = isShouting(TrimmedStatement);
        var Questioning = isQuestioning(TrimmedStatement);

        if (Silent)
        {
            return "Fine. Be that way!";
        } 
        else if (Questioning && Shouting)
        {
            return "Calm down, I know what I'm doing!";
        }
        else if (Questioning)
        {
            return "Sure.";
        }
        else if (Shouting)
        {
            return "Whoa, chill out!";
        }

        return "Whatever.";
    }

    private static bool isSilent(string statement)
    {
        return statement.Length == 0;
    }

    private static bool isQuestioning(string statement)
    {
        return statement.EndsWith("?");
    }

    private static bool isShouting(string statement)
    {
        String UpperCase = statement.ToUpper();
        Boolean HasLetters = (Regex.Matches(UpperCase, @"[A-Z]").Count > 0);
        return statement.Equals(UpperCase) && HasLetters;
    }
}

File: kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {

        if (input.isSilence()) return "Fine. Be that way!"


        val trimmedInput = input.trim()

        val containsLetters = trimmedInput.any { it.isLetter() }
        val isQuestion = trimmedInput.endsWith('?')

        if (containsLetters) {
            val isYelling = trimmedInput == trimmedInput.uppercase()

            if (isYelling && isQuestion) return "Calm down, I know what I'm doing!"
            if (isYelling) return "Whoa, chill out!"
        }
        if (isQuestion) return "Sure."

        return "Whatever."
    }
}

private fun String.isSilence(): Boolean = this.trim().isEmpty()
